src/services/notificationService.ts

```diff
@@ -12,53 +11,36 @@ import {
 
 import { FiscalCode } from "../../generated/backend/FiscalCode";
 import { Installation } from "../../generated/backend/Installation";
-import { PlatformEnum } from "../../generated/backend/Platform";
+import {
+  KindEnum as CreateOrUpdateKind,
+  NotificationHubCreateOrUpdateMessage
+} from "../../generated/messages/NotificationHubCreateOrUpdateMessage";
+import {
+  KindEnum as DeleteKind,
+  NotificationHubDeleteMessage
+} from "../../generated/messages/NotificationHubDeleteMessage";
+import { NotificationHubMessageKindEnum } from "../../generated/messages/NotificationHubMessageKind";
+import {
+  KindEnum as NotifyKind,
+  NotificationHubNotifyMessage
+} from "../../generated/messages/NotificationHubNotifyMessage";
 import { Notification } from "../../generated/notifications/Notification";
 import { SuccessResponse } from "../../generated/notifications/SuccessResponse";
 
+import { QueueClient } from "@azure/storage-queue";
 import { fromNullable } from "fp-ts/lib/Option";
-import {
-  APNSPushType,
-  IInstallation,
-  INotificationTemplate,
-  toFiscalCodeHash
-} from "../types/notification";
-
-/**
- * A template suitable for Apple's APNs.
- *
- * @see https://developer.apple.com/library/content/documentation/NetworkingInternet/Conceptual/RemoteNotificationsPG/CreatingtheNotificationPayload.html
- */
-const APNSTemplate: INotificationTemplate = {
-  body:
-    '{"aps": {"alert": {"title": "$(title)", "body": "$(message)"}}, "message_id": "$(message_id)"}'
-};
-
-/**
- * Build a template suitable for Google's GCM.
- *
- * @see https://developers.google.com/cloud-messaging/concept-options
- */
-const GCMTemplate: INotificationTemplate = {
-  body:
-    '{"data": {"title": "$(title)", "message": "$(message)", "message_id": "$(message_id)", "smallIcon": "ic_notification", "largeIcon": "ic_notification"}}'
-};
-
-// send the push notification only to the last
-// device that set the installationId
-// see https://docs.microsoft.com/en-us/azure/notification-hubs/notification-hubs-push-notification-registration-management#installations
-export const toNotificationTag = (fiscalCode: FiscalCode) =>
-  `$InstallationId:{${toFiscalCodeHash(fiscalCode)}}`;
+import { toFiscalCodeHash } from "../types/notification";
+import { base64EncodeObject } from "../utils/messages";
 
 export default class NotificationService {
-  private notificationHubService: azure.NotificationHubService;
+  private notificationHubQueueClient: QueueClient;
   constructor(
-    private readonly hubName: string,
-    private readonly endpointOrConnectionString: string
+    private readonly queueStorageConnectionString: string,
+    private readonly queueName: string
   ) {
-    this.notificationHubService = azure.createNotificationHubService(
-      this.hubName,
-      this.endpointOrConnectionString
+    this.notificationHubQueueClient = new QueueClient(
+      this.queueStorageConnectionString,
+      this.queueName
     );
```

Comment:
can you check if it is possible to enabled connection keepalive on QueueClient or if is enabled by default ?

Reply:
From the code comment inside `@azure/core-http` seems that keepAlive is `true` by default
```
export interface KeepAliveOptions {
  /*
   * When true, connections will be kept alive for multiple requests.
   * Defaults to true.
   */
  enable: boolean;
}
```
But we can set anyway to `true` for more clear codebase.
```
this.notificationHubQueueClient = new QueueClient(
      this.queueStorageConnectionString,
      this.queueName,
      {
        keepAliveOptions: {
          enable: true
        }
      }
    );
```